typing/includemod.ml

```diff
@@ -922,10 +922,14 @@ let can_alias env path =
 type explanation = Env.t * Error.all
 exception Error of explanation
 
+type application_name =
+  | Anonymous_functor
+  | Full_application_path of Longident.t
+  | Named_leftmost_functor of Longident.t
```

Comment:
I have a hunch that it might be better to keep located identifiers in the error payload. (Something about future programming handling of errors with fixup suggestions etc.) Feel free to act on it or not.